### DISCUSSION TITLE
dhcpv6_relays for ipv6 dhcpv6 relays in vxlans vlan tree

### DISCUSSION THREAD

**Discussion by makzdot** [SUNET/cnaas-nms#541] [General]
**Is your feature request related to a problem? Please describe.**
We CNAAS-NMS has an option for configuring dhcp_relays. because (for example) Juniper has a different config tree for dhcpv6 we have to 'detect' in Jinja if a dhcp relay is ipv4 or ipv6. This makes the jinja2 templates hacky and ugly. 

**Describe the solution you'd like**
Thats why It would be prettier tot just include an option dhcpv6_relays as a list of ipv6 addresses

**Describe alternatives you've considered**
The alternative is to solve it in jinja2 templates. But that can become hacky and cumbersome....at least for a Juniper

**Additional context**
None

**Comment by lunkwill42** [comment]
Alternately, maybe it would be useful to have a generic template filter that can filter any list of IP addresses based on address family?

E.g. (pseudo-code):

```
{{ for addr in filter_ip_family(dhcp_relays, version=6) }}
```

**Comment by makzdot** [comment]
> Alternately, maybe it would be useful to have a generic template filter that can filter any list of IP addresses based on address family?
> 
> E.g. (pseudo-code):
> 
> ```
> {{ for addr in filter_ip_family(dhcp_relays, version=6) }}
> ```

Ah that could probably work let me play with that for a moment. Thanks

**Comment by Qwiko** [comment]
I am not sure how inactive this issue is but, this should be possible using the ipaddress_address filter that is included by default in nms through netutils jinja2_convenience_function.

`{{ "10.1.1.1" | ipaddress_address('version') == 4 }}`


```jinja2
{# In vars: dhcp_relays = [{"host": "10.0.0.1"}, {"host": "2001::0"}] #}

{% for dhcp_relay in dhcp_relays %}
    {% if dhcp_relay.host | ipaddress_address('version') == 6 %}
 ipv6 helper {{ dhcp_relay.host }}
    {% endif %}
{% endfor %}

{# Output: ' ip helper 2001::0' #}
```

One-liner
```jinja2
{% for dhcp_relay in dhcp_relays if dhcp_relay.host | ipaddress_address('version') == 6 %}
 ipv6 helper {{ dhcp_relay.host }}
{% endfor %}
```

There is also possible in jinja to set variables and update them within a for loop and use/reuse the filtered list later.

```jinja2
{% set dhcp6_relays = [] %}
{% for dhcp_relay in dhcp_relays %}
    {% if dhcp_relay.host | ipaddress_address('version') == 6 %}
        {% set _ = dhcp6_relays.append(dhcp_relay) %}
    {% endif %}
{% endfor %}
{# later in the template... #}
{% for dhcp6_relay in dhcp6_relays %}
 ipv6 helper {{ dhcp6_relay.host }}
{% endfor %}
```

Docs: https://netutils.readthedocs.io/en/latest/dev/code_reference/ip/?h=ipaddress_address#netutils.ip.ipaddress_address
